src/bigartm/cooccurrence_dictionary.cc

```diff
@@ -0,0 +1,442 @@
+// Copyright 2017, Additive Regularization of Topic Models.
+
+#include "cooccurrence_dictionary.h"
+
+#include <cstdio>
+#include <unordered_map>
+#include <string>
+#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <map>
+#include <vector>
+#include <assert.h>
+#include <future>
+#include <mutex>
+#include <sstream>
+#include <iomanip>
+
+#include "boost/algorithm/string.hpp"
+#include "boost/filesystem.hpp"
+#include "boost/utility.hpp"
+
+using namespace std;
+using namespace boost;
+
+CooccurrenceBatch::CooccurrenceBatch(int batch_num, const char filemode, const std::string& disk_path) {
+  const int max_name_len = 30;
+  char num_str[max_name_len] = "";
+  sprintf(num_str,"%.3d", batch_num);
```

Comment:
Everywhere use logging:

LOG(ERROR | WARNING | NOTE) << some text without '\n' at the end